dnanexus/dx-streaming-upload
# XVG-8584 resillient log

# JIRA-ticket
https://jira.internal.dnanexus.com/browse/XVG-8584
# Features:
- Copy log from local folder ("~/") to the remote log folder. Try for 3 attempts. If still failed after 3 attempts, backup the log file at the local folder ~/
- add --log-folder to the monitor_run.py script so it aware of the remote log folder
- add --log-name and --log-dsu-name so monitor_run.py knows which log files to pick up 
# Test
- Run the test container
```bash
make base
make build name=xvg-8584
make run name=xvg-8584 dx_token=<your dx token>
```
- Exec into the test docker container
- Wait for the files to be uploaded
- Expect to see the backed up log files in the /root folder as format : /root/dx-stream_cron_run-folder-a.log_1715015367 and /root/monitor_run-folder-a.log_1715015367
- Expect to see these logging in the log file:
```
[proc:919][monitor_runs.py][May 06 2024, 05:09:27 PM (UTC)][WARNING] 1 attempt failed when trying to persist the log file to remote folder /opt/dsu-logs due to error: [Errno 2] No such file or directory: '/opt/dsu-logs/monitor_run-folder-a.log'
[proc:919][monitor_runs.py][May 06 2024, 05:09:27 PM (UTC)][WARNING] 2 attempt failed when trying to persist the log file to remote folder /opt/dsu-logs due to error: [Errno 2] No such file or directory: '/opt/dsu-logs/monitor_run-folder-a.log'
[proc:919][monitor_runs.py][May 06 2024, 05:09:27 PM (UTC)][WARNING] 3 attempt failed when trying to persist the log file to remote folder /opt/dsu-logs due to error: [Errno 2] No such file or directory: '/opt/dsu-logs/monitor_run-folder-a.log'
[proc:919][monitor_runs.py][May 06 2024, 05:09:27 PM (UTC)][WARNING] Failed to persist the log file to the remote folder /opt/dsu-logs after 3 attempts
[proc:919][monitor_runs.py][May 06 2024, 05:09:27 PM (UTC)][WARNING] Backed up the log file at /root/monitor_run-folder-a.log_1715015367
[proc:919][monitor_runs.py][May 06 2024, 05:09:27 PM (UTC)][WARNING] Backed up the dx-stream-cron log file at /root/dx-stream_cron_run-folder-a.log_1715015367
```
